modules/xfd.ts

```diff
@@ -0,0 +1,2218 @@
+		return 'Nominate article for deletion or move';
+	}
+
+	getFieldsetLabel() {
+		return 'Miscellany for deletion';
+	}
+
+	generateFieldset(): quickFormElement {
+		this.fieldset = super.generateFieldset();
+		this.fieldset.append({
+			type: 'checkbox',
+			list: [
+				{
+					label: 'Wrap deletion tag with <noinclude>',
+					value: 'noinclude',
+					name: 'noinclude',
+					tooltip: 'Will wrap the deletion tag in &lt;noinclude&gt; tags, so that it won\'t transclude. Select this option for userboxes.'
+				}
+			]
+		});
+		if ((mw.config.get('wgNamespaceNumber') === 2 /* User: */ || mw.config.get('wgNamespaceNumber') === 3 /* User talk: */) && mw.config.exists('wgRelevantUserName')) {
+			this.fieldset.append({
+				type: 'checkbox',
+				list: [
+					{
+						label: 'Notify owner of userspace (if they are not the page creator)',
+						value: 'notifyuserspace',
+						name: 'notifyuserspace',
+						tooltip: 'If the user in whose userspace this page is located is not the page creator (for example, the page is a rescued article stored as a userspace draft), notify the userspace owner as well.',
+						checked: true
+					}
+				]
+			});
+		}
+		this.appendReasonArea();
+		return this.fieldset;
+	}
+
+	preprocessParams() {
+		this.params.userspaceOwner = mw.config.get('wgRelevantUserName');
+	}
+
+	evaluate() {
+		let tm = super.evaluate();
+		tm.add(this.determineDiscussionPage, [])
+		tm.add(this.tagPage, [this.determineDiscussionPage]);
+		tm.add(this.addToList, [this.determineDiscussionPage]);
+		tm.add(this.createDiscussionPage, [this.determineDiscussionPage]);
+		tm.add(this.fetchCreatorInfo, []);
+		tm.add(this.notifyCreator, [this.fetchCreatorInfo]);
+		tm.add(this.notifyUserspaceOwner, [this.fetchCreatorInfo]);
+		tm.add(this.addToLog, [this.notifyCreator, this.notifyUserspaceOwner]);
+		tm.execute().then(() => this.redirectToDiscussion());
+	}
+
+	determineDiscussionPage() {
+		let params = this.params;
+		let wikipedia_api = new Morebits.wiki.api('Looking for prior nominations of this page', {
+			'action': 'query',
+			'list': 'allpages',
+			'apprefix': 'Miscellany for deletion/' + Morebits.pageNameNorm,
+			'apnamespace': 4,
+			'apfilterredir': 'nonredirects',
+			'aplimit': 'max' // 500 is max for normal users, 5000 for bots and sysops
+		});
+		return wikipedia_api.post().then((apiobj) => {
+			var xmlDoc = apiobj.responseXML;
+			var titles = $(xmlDoc).find('allpages p');
+
+			// There has been no earlier entries with this prefix, just go on.
+			if (titles.length <= 0) {
+				params.numbering = params.number = '';
+			} else {
+				var number = 0;
+				for (var i = 0; i < titles.length; ++i) {
+					var title = titles[i].getAttribute('title');
+
+					// First, simple test, is there an instance with this exact name?
+					if (title === 'Wikipedia:Miscellany for deletion/' + Morebits.pageNameNorm) {
+						number = Math.max(number, 1);
+						continue;
+					}
+
+					var order_re = new RegExp('^' +
```

Comment:
same comment as on the javascript - suggest creating the RegExp object outside of the for loop

Reply:
You're right. But this [predates the initial github import 10 years ago](https://github.com/azatoth/twinkle/commit/71267fd4f1a0057c01ddc9ec5eb8ea40cf2a75be#diff-b0bf7dacc07995995276a293b020a474a08313a5f921eb1d7f623fdcd3b6394cR406)!